synapse/lib/queue.py

```diff
@@ -40,6 +42,77 @@ async def slice(self):
         self.event.clear()
         return retn
 
+class Queue:
+    '''
+    An asyncio Queue with batch methods and graceful close.
+    '''
+    def __init__(self, maxsize=None):
+        self.q = asyncio.Queue(maxsize=maxsize)
+        self.closed = False
+
+    async def close(self):
+        await self.q.put(s_common.novalu)
+        self.closed = True
+
+    async def put(self, item):
+
+        if self.closed:
+            mesg = 'The Queue has been closed.'
+            raise s_exc.BadArg(mesg=mesg)
+
+        return await self.q.put(item)
```

Comment:
This returns None; whereas the puts() as implemented returns size of input. couldn't the caller just call len on the size of the items they put in? they should be the same.